sunpy/coordinates/tests/test_sun.py

```diff
@@ -501,7 +502,7 @@ def test_carrington_rotation_time_longitude_numpy(crot, longitude, crot_fraction
                              (2000.5, 180),
 ])
 def test_carrington_rotation_time_longitude_err(crot, longitude):
-    with pytest.raises(ValueError):
+    with pytest.raises(ValueError, match=re.escape("Carrington longitude(s) must be > 0 degrees and <= 360 degrees.") + "|" + re.escape("Carrington rotation number(s) must be integral if `longitude` is provided.")):
```

Comment:
Maybe shrink this down to "Carrington rotation number(s) must be integral if `longitude` is provided."?

Reply:
On doing this, the following error occurs : 

During handling of the above exception, another exception occurred:
sunpy/coordinates/tests/test_sun.py:505: in test_carrington_rotation_time_longitude_err
    with pytest.raises(ValueError, match=re.escape(re.escape("Carrington rotation number(s) must be integral if `longitude` is provided."))):
E   AssertionError: Regex pattern did not match.
E    Regex: 'Carrington\\\\\\ rotation\\\\\\ number\\\\\\(s\\\\\\)\\\\\\ must\\\\\\ be\\\\\\ integral\\\\\\ if\\\\\\ `longitude`\\\\\\ is\\\\\\ provided\\\\\\.'
E    Input: 'Carrington longitude(s) must be > 0 degrees and <= 360 degrees.'

Reply:
You should add the error message to the pytest parametrize and then have it check for that in the regex, since the code is raising different types of errors depending on the input.

Reply:
@nabobalis Done. Thank You.